tests/unit/get-parser-plugin-by-parser-name.js

```diff
@@ -0,0 +1,65 @@
+import { ConfigError } from "../../src/common/errors.js";
+import { getParserPluginByParserName } from "../../src/main/parser-and-printer.js";
+import createPlugin from "../config/utils/create-plugin.cjs";
+
+describe("unit tests for getParserPluginByParserName", () => {
+  const getMockPlugins = (name) => [
+    { name: "prettier-plugin-do-not-have-parsers" },
+    createPlugin({ name, print: () => "test" }),
+  ];
+
+  const { env } = process;
+
+  beforeEach(() => {
```

Comment:
Normally when mutating env vars, I would call `jest.resetModules()` in `afterEach`. The test runner doesn't expose a `resetModules()` function as far as I could tell, which is why I didn't do that here. I read [documentation](https://github.com/nicolo-ribaudo/jest-light-runner) saying each test file is isolated, but this would still fail unnecessarily if someone moved `process.env.PRETTIER_TARGET` to the parent scope of `getParserPluginByParserName`. That saidf, it's passing now. Let me know if there's an alternative approach & I'm happy to fix!